samples/sqlservices/src/controllers/mainController.ts

```diff
@@ -701,9 +708,139 @@ export default class MainController implements vscode.Disposable {
+		const barChart = view.modelBuilder.chart<azdata.BarChartConfiguration>()
+			.withProps({
+				chartId: 'barChart1',
+				chartType: 'bar',
+				configuration: chartExamples.barConfig,
+				width: '500px',
+				height: '300px'
+			}).component();
+
+		const horizontalBarChart = view.modelBuilder.chart<azdata.BarChartConfiguration>() // BarChartConfiguration is used for both vertical and horizontal bar charts
+			.withProps({
+				chartId: 'horizBarChart1',
+				chartType: 'horizontalBar',
+				configuration: chartExamples.horizontalBarConfig,
+				width: '500px',
+				height: '300px'
+			}).component();
+
+		const lineChart = view.modelBuilder.chart<azdata.LineChartConfiguration>()
+			.withProps({
+				chartId: 'lineChart1',
+				chartType: 'line',
+				configuration: chartExamples.lineConfig,
+				width: '500px',
+				height: '300px'
+			}).component();
+
+		const pieChart = view.modelBuilder.chart<azdata.PieChartConfiguration>()
+			.withProps({
+				chartId: 'pieChart1',
+				chartType: 'pie',
+				configuration: chartExamples.pieConfig,
+				width: '300px',
+				height: '300px'
+			}).component();
+
+		const doughnutChart = view.modelBuilder.chart<azdata.PieChartConfiguration>() // PieChartConfiguration is used for both pie charts and doughnut charts
+			.withProps({
+				chartId: 'doughnutChart1',
+				chartType: 'doughnut',
+				configuration: chartExamples.doughnutConfig,
+				width: '400px',
+				height: '400px'
+			}).component();
+
+		const scatterplot = view.modelBuilder.chart<azdata.ScatterplotConfiguration>()
+			.withProps({
+				chartId: 'scatterplot1',
+				chartType: 'scatter',
+				configuration: chartExamples.scatterConfig,
+				width: '400px',
+				height: '400px'
+			}).component();
+
+		const bubbleChart = view.modelBuilder.chart<azdata.BubbleChartConfiguration>()
+			.withProps({
+				chartId: 'bubbleChart1',
+				chartType: 'bubble',
+				configuration: chartExamples.bubbleConfig,
+				width: '500px',
+				height: '500px'
+			}).component();
 
-	//#region Registrations
+		const polarChart = view.modelBuilder.chart<azdata.PolarAreaChartConfiguration>()
+			.withProps({
+				chartId: 'polarChart1',
+				chartType: 'polarArea',
+				configuration: chartExamples.polarConfig,
+				width: '500px',
+				height: '500px'
+			}).component();
+
+		const radarChart = view.modelBuilder.chart<azdata.RadarChartConfiguration>()
+			.withProps({
+				chartId: 'radarChart1',
+				chartType: 'radar',
+				configuration: chartExamples.radarConfig,
+				width: '500px',
+				height: '500px'
+			}).component();
+
+		const button = view.modelBuilder.button()
+			.withProps({
+				label: 'Click to change bar chart data'
+			}).component();
+
+		button.onDidClick(async () => {
+			// To update data, a new config object must be created and passed.
+			// If the existing one is updated, it's detected as the same object, and "saves" the effort of send propertyChanged events.
```

Comment:
I don't love having to recreate the entire configuration (and therefore set options when only needing to set new data) instead of just setting the data, so I'm soliciting some feedback for some options:

1. Keep it this way.
2. Separate out data from options ([prototyping in this commit](https://github.com/microsoft/azuredatastudio/commit/b6ed79aad7408fe0fd2e19289da91eaf0b7e74d4)), but TypeScript isn't enforcing the TData-TOption pairings the way I thought it would (meaning someone could pass PieChartOptions with BarChartData).  Not the worst, but I'm looking for solutions to that...
3. Edit data in-place (similar to code sample), then add a `Redraw()` method to the chart component to have it pick up the changes.

Thoughts?  Other ideas?

Reply:
I like how option 3 sounds the best, if the `Redraw()` is done by the component and the extension doesn't need to call it. The table component lets you only update the data without needing to send all the properties, so makes sense to support doing something similar for charts.

Reply:
As requested, I've added examples of what it would look like to update the data in each choice.

Reply:
@kisantia I'll poke through the table component code to see if I can do something the same way.

Reply:
Thanks for adding the examples! I was confused earlier and thought that option 3 was what option 2 actually is. That's how the other model view components, like table, handle updating data, so I like that part of it. It doesn't seem that bad to me to throw an error if the wrong data type is passed in.

Reply:
+1 on doing it like table does, being consistent with how we handle data seems best long term unless we come up with a clearly better approach.

Reply:
Okay, I've updated the design to #2 and also fixed it so that it won't allow mismatched data and options types (which typescript made a bit annoying).